### PR TITLE
Use ticker price to determine sell price, add some more exception handling, and some refactoring

### PR DESCRIPTION
**Pull request by ogzd**
We should use latest ask price when deciding the sell price in case the ask price can be higher than the current profit. So we can earn more money $$.

This PR contains more changes regarding to exception handling. Some commented-out hacky code where it is useful to change the timestamp couple of seconds back in time to keep Binance happy.

I have also moved the Thread.sleep call to the end of the round, so we can still keep tracking the existing open buy orders even when the Binance API returning us noisy errors. For some cases, where the order or its status is not available, we assume the order doesn't exist and remove it from the current queue.

### CODE REVIEW

File: coinmx/src/main/java/com/coinmx/strategy/StrategyExecutor.java [hunk 1]
```diff
@@ -152,18 +169,30 @@ private Runnable getSellTask(String orderId) {
...
+                    LOG.error("Ticker data not available for order id {}", orderId);
+                } catch (RuntimeException e) {
+                    LOG.error("Runtime exception caught", e);
+                }
+
+                // Sleep 10 seconds before checking the status of the order again.
+                try {
+                    Thread.sleep(TimeUnit.SECONDS.toMillis(10));
+                } catch (InterruptedException e) {
+                    numCurrentOpenBuys.decrementAndGet(); // Why?
```
Comment (ogzd) [inline coinmx/src/main/java/com/coinmx/strategy/StrategyExecutor.java]: Interrupt doesn't break the loop. Why are we decrementing the counter in this case? This is an open question, I didn't modify the previous logic for this exception handling but IMHO seems wrong.

File: coinmx/src/main/java/com/coinmx/exceptions/OrderStatusNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.coinmx.exceptions;
+
+public class OrderStatusNotFoundException extends Exception {
+    public OrderStatusNotFoundException(Exception e) {
+        super(e);
+    }
+
+    public OrderStatusNotFoundException() {
+        super("Order status not found");
```
Comment (ortschun) [inline coinmx/src/main/java/com/coinmx/exceptions/OrderStatusNotFoundException.java]: Order status not found ile, no such order arasindaki fark ne 💃

File: coinmx/src/main/java/com/coinmx/strategy/StrategyExecutor.java (hunk 1 above)
Comment (ortschun) [reply coinmx/src/main/java/com/coinmx/strategy/StrategyExecutor.java]: normalde burda sanirim bunu decrement edip, thread in interrupted flag ini tekrar set edip, interrupted i yukariya firlatmamiz lazim. O isi yapmiyoruz su an. cok da onemli degil cunku gelecek bir sey degil ama aklimizda bulunsun

File: coinmx/src/main/java/com/coinmx/strategy/StrategyExecutor.java [hunk 3]
```diff
@@ -84,13 +88,21 @@ public void execute() {
...
+                        "Current Open Buys: {}, Max buys at once: {}", numCurrentOpenBuys.get(), maxBuysAtOnce);
                 return;
             }
             try {
                 Exchange exchange = orderManager.getExchange();
 
                 AccountInfo accountInfo = exchange.getAccountService().getAccountInfo();
+//                BinanceAccountService accountService = (BinanceAccountService) exchange.getAccountService();
+//                BinanceAccountInformation acc = accountService.account(1000000L, System.currentTimeMillis() - 2000);
+//                List<Balance> balances = acc.balances.stream()
```
Comment (ortschun) [inline coinmx/src/main/java/com/coinmx/strategy/StrategyExecutor.java]: Burdaki asil sikinti, trade yaparken de bu patlayacak, onu gecirmemiz lazim. Asil cozum buna, adamlar SNAPSHOT versiyonunu cikartmislar apilerinin. Ona depend edip, son degisiklikleri alip, metadata olarak recv parametresi gecirmek olacak. onu yapmak lazim seninkini calistirabilmek icin.

File: coinmx/src/main/java/com/coinmx/exceptions/OrderStatusNotFoundException.java (hunk 2 above)
Comment (ogzd) [reply coinmx/src/main/java/com/coinmx/exceptions/OrderStatusNotFoundException.java]: birinde order yok, digerinde order status yok :D

**Pull request merged**